### PR TITLE
update `tsconfig.json` appropriate for node>=14

### PR DESCRIPTION
**Pull request by KATT**
- be explicit that we only support node `>=14`
- any downsides on this? i guess anyone using trpc already is on node>=14

https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-14

### CODE REVIEW

File: packages/server/package.json [hunk 1]
```diff
@@ -10,6 +10,9 @@
     "url": "git+https://github.com/trpc/trpc.git",
     "directory": "packages/server"
   },
+  "engines": {
+    "node": ">=14"
```
Comment (mmkal) [inline packages/server/package.json:14]: Most users won't be affected by this but it is technically a breaking change, is the plan to do a major release?

Comment (KATT) [reply packages/server/package.json:14]: I actually didn't plan to - I never intended to support node `12.x` in the first place and that tsconfig was accidental. 

That said, you're right, it is a major change. I'm going to check if the lib currently even works in node 12 and then decide what to do.

Comment (KATT) [reply packages/server/package.json:14]: Unfortunately, trpc seems to work with `12.x` which means some people might be using it so I might just merge #886 and wait to do this change until next major

**Comment by KATT** [comment]
Closing in favor of #886 - we'll have to do this in next major

**Pull request closed without merging**